Topics/Tech_Stacks/ORM.md

```diff
@@ -0,0 +1,133 @@
+
+const Student = sequelize.define('Student', {
+    id: {
+        type: DataTypes.INTEGER,
+        primaryKey: true,
+        autoIncrement: true
+    },
+    username: {
+        type: DataTypes.STRING,
+        allowNull: false,
+    },
+    student_number: {
+        type: DataTypes.INTEGER,
+        allowNull: false,
+    },
+});
+
+await Student.sync({force:true}); // Ensures table is created in DB
+
+await Student.create({
+    username: 'sidiki',
+    student_number: 6789998212
+})
+
+await sequelize.close()
+```
+
+```js
+// The following is the equivalent code using RAW SQL more heavily
+import { Client } from 'pg'
+const client = new Client() // Server Details go here
+await client.connect()
+
+const createTableQuery = await client.query(
+    ```
+    Create TABLE Student (
+        id SERIAL PRIMARY KEY,
+        username varchar(50),
+        student_number integer
+    );
+    ```
+);
+
+const student_username = "sidiki" 
+const student_number = 6789998212 
+
+const insertQuery = await client.query(
+    ```
+    Insert into Student (username, student_number) Values 
+    ('${student_username}', ${student_number});
+    ```
+)
+
+await client.end()
+
+```
+<!--
+<p align="center">
+      <img  src="https://github.com/learning-software-engineering/learning-software-engineering.github.io/assets/95612717/b6111831-1240-42f2-868f-76a08647c1db" width="80%" height="50%" display="block">
+</p>
+<p align="center">
+      <img src="https://github.com/learning-software-engineering/learning-software-engineering.github.io/assets/95612717/0faaecfe-2acd-48cc-a5c3-90b89f8833d0" width="80%" height="50%" display="block">
+</p>
+-->
+
+These screenshots use the following packages for the ORM and raw SQL usage examples, respectively:
+1. https://sequelize.org/ 
+2. https://node-postgres.com/ 
+
+Looking at the two examples, despite the fact that a similar structure of steps is performed, we can immediately notice the differences between the 2 strategies of interacting with databases. The raw SQL strategy makes heavy use of strings, after all these strings are directly passed to the database to run the SQL code. On the other hand, the ORM method sets up object relation and creates an instance as if it were another object within our code.
+
+
+## Should I Use One!?
+
+Although the implementation looks different, you can still do everything that ORMS have to offer with just raw SQL queries paired with a database driver and vice versa, so then which one should you actually use? 
+As with many choices in the software engineering world (like which language to use, or which framework to code in) there are no clear-cut answers. It is your responsibility to understand the advantages and disadvantages of the two options and make an informed decision on how you will be interacting with your database. Let us discuss some of these tradeoffs so that you can make a more informed decision, whichever one you choose to use.
+
+### Advantages:
```

Comment:
Great comparison of the advantages and disadvantages! They help to highlight the important points to consider when choosing an appropriate approach. 

One small thing, it might be helpful to clarify what 'reduction' in performance precisely means. Is it speed? or storage? or some other metric?

Reply:
Yes im specifically talking about 'speed', since SQL queries are often automatically optimised by the database engine so they can compute and return faster, I should clarify this!